test-suite/js/src/test/scala/org/scalajs/testsuite/utils/HTMLRunner.scala

```diff
@@ -0,0 +1,136 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.utils
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSName
+
+import scala.collection.mutable
+
+import scala.concurrent.{Future, Promise}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import org.scalajs.testinterface.ScalaJSClassLoader
+
+import sbt.testing._
+
+object HTMLRunner extends js.JSApp {
+  private val classLoader = new ScalaJSClassLoader(js.Dynamic.global)
+
+  private val logger = new Logger {
+    val ansiCodesSupported = false
+    def error(msg: String): Unit = UI.log(msg, "error")
+    def warn(msg: String): Unit = UI.log(msg, "warn")
+    def info(msg: String): Unit = UI.log(msg, "info")
+    def debug(msg: String): Unit = UI.log(msg, "debug")
+    def trace(t: Throwable): Unit = UI.log(t.toString, "error")
+  }
+
+  private object EventCounter extends EventHandler {
+    val counts = mutable.Map.empty[Status, Int].withDefaultValue(0)
+    def handle(event: Event): Unit = counts(event.status) += 1
+  }
+
+  def main(): Unit = {
+    System.setProperty("scalajs.testsuite.testtag", "testtag.value")
+    System.setProperty("scalajs.browser", "true")
+    System.setProperty("scalajs.typedarray", "true")
```

Comment:
This isn't for every browser, but the browsers we test in. So yes, it was an assumption we had in our runners before.